builders/tslice.py

```diff
@@ -138,24 +135,40 @@ def build_df(
     if verify(input_yaml):
         # loop over datasets
         df_parts = []
+        entries = {}
         for dataset in input_yaml.keys():
             ds_globals = {}
             # get a list of keys that are common to all files in the dataset
             for g in input_yaml[dataset].keys():
-                if 'data_sources' not in g:
+                if 'data_sources' not in g and 'ensemble' not in g:
                     ds_globals[g] = input_yaml[dataset][g]
+            # loop over ensemble members, if they exist
+            if 'ensemble' in input_yaml[dataset].keys():
+                for member in input_yaml[dataset]['ensemble']:
+                    filelist = get_asset_list(member['glob_string'], depth=0) 
+                    member.pop('glob_string')
+                    for f in filelist:
+                        entries[f] = member
```

Comment:
Use the `defaultdict` capabilities:

```suggestion
                        entries[f].update(member)
```